### PR TITLE
Migrated to msal-browser auth

### PR DESCRIPTION
**Pull request by chormaz**
# Description
Migrated auth from msal to azure-msal-browser.

Completes: [AB#90036](https://statoildeveloper.visualstudio.com/b6d97460-2e53-47ae-aad4-b41396d90828/_workitems/edit/90036)

# Checklist:

- [X] I have performed a self-review of my own code.
- [X] I have linked my DevOps task using the AB# tag.
- [X] My code is easy to read, and comments are added where needed.

### CODE REVIEW

**Comment by bratli** [comment]
LGTM, but I have not worked with auth before, so don't know if we should get anyone who has? Or maybe you have already discussed this?

File: src/auth/AuthService.tsx [hunk 1]
```diff
@@ -35,60 +36,185 @@ export interface IAuthService {
...
 
-    logout(): void {
-        localStorage.clear();
-        sessionStorage.clear();
-        window.location.assign('https://myapps.microsoft.com/');
+    /**
+     * Handles the response from a popup or redirect. If response is null, will check if we have any accounts and attempt to sign in.
+     * @param response
+     */
+    handleResponse(response: AuthenticationResult | null) {
```
Comment (VRiveland) [inline src/auth/AuthService.tsx:148]: Return type void?

File: src/auth/AuthService.tsx [hunk 2]
```diff
@@ -35,60 +36,185 @@ export interface IAuthService {
...
 
-    handleRedirectCallback(): void {
-        try {
-            this.authInstance.handleRedirectCallback((err /* response */) => {
-                if (err) throw new AuthenticationError(err.message);
+    /**
+     * Calls ssoSilent to attempt silent flow. If it fails due to interaction required error, it will prompt the user to login using popup.
+     * @param request
+     */
+    attemptSsoSilent() {
```
Comment (VRiveland) [inline src/auth/AuthService.tsx:164]: Return type void?

File: src/auth/AuthService.tsx [hunk 3]
```diff
@@ -110,23 +236,26 @@ export default class AuthService implements IAuthService {
                     'consent_required',
                     'interaction_required',
                     'login_required',
-                ].indexOf(authError.errorCode) !== -1
+                ].indexOf((authError as AuthError).errorCode) !== -1
             ) {
                 console.log('Consent issue'); // TODO: remove once bug is fixed
```
Comment (VRiveland) [inline src/auth/AuthService.tsx:241]: Could you remove?

File: src/auth/AuthService.tsx [hunk 4]
```diff
@@ -110,23 +236,26 @@ export default class AuthService implements IAuthService {
...
-                this.aquireConcent(resource);
-                console.log('Consent accuired'); // TODO: remove once bug is fixed
+                await this.myMSALObj.acquireTokenRedirect({
+                    scopes: [resource],
+                });
+
+                console.log('Consent acquired'); // TODO: remove once bug is fixed
```
Comment (VRiveland) [inline src/auth/AuthService.tsx:246]: Could you remove?

**Review by VRiveland — APPROVED** [review]

**Pull request merged**